tests/unit/conftest.py

```diff
@@ -10,62 +10,113 @@
         monkeypatch.setattr(_creation_management, '_cache_kvs_by_id', {})
         monkeypatch.setattr(_creation_management, '_cache_kvs_by_name', {})
         monkeypatch.setattr(_creation_management, '_cache_rq_by_id', {})
         monkeypatch.setattr(_creation_management, '_cache_rq_by_name', {})
 
-        # Verify that the environment variable is set correctly
+        # Verify that the test environment was set up correctly.
         assert os.environ.get('CRAWLEE_STORAGE_DIR') == str(tmp_path)
+        assert service_locator._service_locator.configuration_was_set is False
+        assert service_locator._service_locator.storage_client_was_set is False
+        assert service_locator._service_locator.event_manager_was_set is False
 
-    return reset
+    return _prepare_test_env
+
+
+@pytest.fixture
+def cleanup_test_env() -> Callable[[], None]:
+    """Clean up and reset global state after a test has completed.
+
+    This fixture ensures that any modifications to the global state during a test are undone
+    after the test finishes. Restoring the environment for the subsequent tests.
+
+    Returns:
+        A callable that cleans up the test environment.
+    """
+
+    def _cleanup_test_env() -> None:
+        # Reset the flags in the service locator to indicate no services are explicitly set.
+        service_locator._service_locator._configuration_was_set = False
+        service_locator._service_locator._storage_client_was_set = False
+        service_locator._service_locator._event_manager_was_set = False
+
+    return _cleanup_test_env
 
 
 @pytest.fixture(autouse=True)
-def _isolate_test_environment(reset_globals: Callable[[], None]) -> None:
-    """Isolate tests by resetting the storage clients, clearing caches, and setting the environment variables.
+def _isolate_test_environment(
+    prepare_test_env: Callable[[], None],
+    cleanup_test_env: Callable[[], None],
+) -> Generator[None, None, None]:
+    """Isolate the testing environment by resetting global state before and after each test.
 
-    The fixture is applied automatically to all test cases.
+    This fixture ensures that each test starts with a clean slate and that any modifications during the test
+    do not affect subsequent tests. It runs automatically for all tests.
 
     Args:
-        monkeypatch: Test utility provided by pytest.
-        tmp_path: A unique temporary directory path provided by pytest for test isolation.
-    """
+        prepare_test_env: Fixture to prepare the environment before each test.
+        cleanup_test_env: Fixture to clean up the environment after each test.
 
-    reset_globals()
+    Yields:
+        None. This fixture works as a setup and teardown mechanism.
+    """
+    prepare_test_env()
+    yield
+    cleanup_test_env()
 
 
 @pytest.fixture
 def memory_storage_client(tmp_path: Path) -> MemoryStorageClient:
-    cfg = Configuration(
+    return MemoryStorageClient(
```

Comment:
Maybe not, but shouldn't we also plug this one into the `service_locator` here?

Reply:
I don't think this is necessary, as this fixture is intended for testing the memory storage client (and its resource clients). Maybe it doesn't belong here.

Reply:
Huh, probably not :slightly_smiling_face:

Reply:
I'd say I can move the fixture somewhere to `tests/memory_storage_client`